src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -32,193 +32,668 @@
+                    overwrittenMapping.setTargetField(fieldMapping.targetFieldAPIName);
+
+                    queuedMetadataTypesForDeploy.add(overwrittenMapping.getMetadataRecord());
+                } else {
+                    // Check if we have a conflict and update the target field based on the help text mapping
+                    if (fieldMapping.hasBadDefaultValues(existingDefaultFieldMapping)) {
+                        fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                        fieldMapping.setObjectMapping(existingDefaultObjectMapping);
+                        fieldMapping.setTargetField(fieldMapping.targetFieldAPIName);
+                    } else {
+                        fieldMapping.setFieldMappingSet(MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                        fieldMapping.setObjectMapping(existingDefaultObjectMapping);
+                    }
+
+                    queuedMetadataTypesForDeploy.add(fieldMapping.getMetadataRecord());
+                }
+            }
+
+            fieldMappingsByFieldMappingSetDeveloperName.remove(fieldMappingKey);
+        }
+
+        createAndEnqueueNonHelpTextMappings(fieldMappingsByFieldMappingSetDeveloperName);
+        createAndEnqueueMigratedCustomFieldMappingSet();
     }
 
-    /**
-     * @description Collect all Help Text field mappings and compare against default Field Mapping Metadata 
-     * Type records reserve all non-standard NPSP Help Text mappings. What to do with reserved list of Help 
-     * Text mappings TBD.
+    /*******************************************************************************************************
+     * @description Go through all fields in Data Import object and populate relevant lists from help text
+     * mappings
      */
-    public void buildCustomMetadataRecordsFromHelpText(BDI_MigrationMappingHelper helper) {
+    public void populateListsFromHelpTextMappings() {
 
-        Map<String, BDI_HelpTextCMT.MappingDetail> listHelpTextFieldDetails =
-                helper.getAllMappedHelpTextFields();
+        badHelpTextMappings.clear();
+        dataImportFieldMappings.clear();
 
-        Map<String, Data_Import_Object_Mapping__mdt> existingObjectMappings =
-                helper.getDefaultObjectMappings(DEFAULT_OBJECT_MAPPING_SET.Id);
+        // TODO: Need to move this into the helper so we can mock the result
+        // Replace stubbedMethod getDataImportFieldMappingsFromHelpText
+        // When this method was in the helper, we had another get method
+        // used by this utility class on line 238
+        // i.e. List<DataImportFieldMapping> helpTextFieldMappings = helper.getDataImportFieldMappingsFromHelpText();
+        Map<String, Schema.DescribeFieldResult> describeFieldResultsByFieldName =
+            UTIL_Describe.getAllFieldsDescribe(UTIL_Namespace.StrAllNSPrefix(DATA_IMPORT_OBJECT_NAME));
```

Comment:
Yep, this could be `helper.getAllFieldsDescribe(DATA_IMPORT_OBJECT_NAME)`

Reply:
@kenlewis-sfdo - I was trying this out yesterday and I don't think this would actually be useful. Since that would need to return a Map<String, Schema.DescribeFieldResult>. Am I wrong in that there isn't a way for us to mock or otherwise construct instances of Schema.DescribeFieldResults?